artemis-server/src/main/java/org/apache/activemq/artemis/utils/XMLConfigurationUtil.java

```diff
@@ -44,7 +44,6 @@ public static final Double getDouble(final Element e,
          validator.validate(name, val);
          return val;
       } else {
-         validator.validate(name, def);
```

Comment:
???

Reply:
This is about validating the default value.

If the user didn't make a selection, there's no need to validate the default value of a selection.


I have been through specifically a case where device-block-size (a parameter that I will add) would be null if no selection, but in case there's a selection it must abide by the rules.

The default value == null would be allowed. 

There's no need to validate the default return value.

Reply:
@michaelandrepearce any objection?

This should not be an issue, but I wanted to make sure.

Reply:
@michaelandrepearce I need this fix in as I'm doing tests on a device that has block-size != 4096.

if you have any objections I can send further changes.

Reply:
Surely though it would be valid for someone to set a default value, as such whilst a bit backwards, this would actually be (from unit testing) validate that the validators set allow for the default value. If they dont and they fail, then the validator isnt correct.

Reply:
@michaelandrepearce I'm confused if you want me to just add a test or add the validator back?

Reply:
Validator back

Reply:
@michaelandrepearce well.. I don't see a point on that.. really.

It's moot, it's dead code.

Reply:
Fair enuf. I see your point